src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1501,14 +1502,18 @@ internal static void RemoveNestedModuleFunctions(PSModuleInfo module)
             }
 
             input.Clear();
-            input.AddRange(output);
+            foreach (var fnInfo in output)
+            {
+                input.Add(fnInfo);
+            }
         }
 
-        private static void SortAndRemoveDuplicates<T>(List<T> input, Func<T, string> keyGetter)
+        private static void SortAndRemoveDuplicates<T>(ConcurrentBag<T> input, Func<T, string> keyGetter)
         {
             Dbg.Assert(input != null, "Caller should verify that input != null");
 
-            input.Sort(
+            var inputList = new List<T>(input);
```

Comment:
Because `ConcurrentBag` does not support `.Sort` in the same way, I create a temporary List for it. This should not have a performance implication because the ConcurrentBag `input` gets cleared and re-populated anyway in the code below